### DISCUSSION TITLE
Single Screen to be available for non-logged in users?

### DISCUSSION THREAD

**Discussion by sghriyadh** [Budibase/budibase#10563] [Help] [answered]
Hi 

i have an app which is currently used by logged in users 
it has a search page which shows results.

Can i make a copy of this page"screen" and set it to public so non logged in users can search and see details of rows? 

i am on self hosted version

**Comment by ConorWebb96** [comment, marked as answer]
Hey @sghriyadh,

I don't see why this wouldn't be possible. You are able to duplicate screens.
![Screenshot 2023-05-12 at 13 27 56](https://github.com/Budibase/budibase/assets/126772285/3acbaead-91d6-4036-a456-069b770ca609)

From here you can then set the lower roles who can view this screen.
![Screenshot 2023-05-12 at 13 28 58](https://github.com/Budibase/budibase/assets/126772285/d42b2f1e-4f22-4fdd-85b1-eb8d25b38e0b)

Then you can add a url for this page to allow them to easily get there.
![Screenshot 2023-05-12 at 13 28 41](https://github.com/Budibase/budibase/assets/126772285/330aefa3-ac57-4d34-a65d-f0dc64caeec8)

You will need to make sure that any conditional validation within the screen is changed to public, for example: if you have a component which is hidden for certain users groups this will also need amended so it can be see for public users.

Hopefully this makes sense and is helpful.

**Comment by sghriyadh** [comment]
![16839554352927150635313411395047](https://github.com/Budibase/budibase/assets/130876667/051d32a1-1488-4d11-8703-1c87f2026cea)

I duplicated as public and opened in a new browser without logging in. It gives the session error as shown in the video while loading page.

Components do not have privacy settings as i see. Only screens have options for public .

Is there something that i missed?

**Reply by ConorWebb96** [reply]
This seems to be an issue with your authentication, have you updated the tables you are using within that page?

You'll need to update the permissions to read and possibly write depending on the functionality you have on that page. (I've supplied screenshots below to help with understanding where and how to do this.)
<img width="1071" alt="Screenshot 2023-05-15 at 08 37 08" src="https://github.com/Budibase/budibase/assets/126772285/1bc09e72-adb4-46fb-84cc-408bd8ce55dd">

<img width="638" alt="Screenshot 2023-05-15 at 08 38 11" src="https://github.com/Budibase/budibase/assets/126772285/67298008-cb82-478f-a944-2c3cf782d8a2">

Hopefully this helps.

**Comment by saz33m1** [comment]
Has the granular "Read/Write" access for Tables, been deprecated? I only see this 
<img width="1904" alt="image" src="https://github.com/user-attachments/assets/98a6da93-48c6-471a-b77c-33011552b324">

**Reply by Kry9toN** [reply]
same issue

**Reply by mikesealey** [reply]
Hey @saz33m1 @Kry9toN 

Read and Write access has been put together under the single banner of "Access Control" in v3. 
![image](https://github.com/user-attachments/assets/300e536d-1fcf-4ff6-aed9-f15a7059bd46)

You can grant Read access to specific roles by creating a view of the data, specifying which columns you want to allow the user to view, and then setting them to Read Only.

Create a view
![image](https://github.com/user-attachments/assets/cfc222b8-667f-4435-9d51-7d73e4bd86a3)

Set that view to the role you wish (public, in your case) and specify the columns to be read only or hidden according to your needs. 
![image](https://github.com/user-attachments/assets/f9c345d9-d34a-4a80-af1b-46685a5be85a)

**Reply by saz33m1** [reply]
@mikesealey Appreciate you taking the time to explain views and read-only access. I have a use case where I want a public form to write to a table. More details here: https://github.com/Budibase/budibase/discussions/14971

**Reply by aowdnmp** [reply]
Btw, I tried and with views you cannot make readonly display columns or required fields...

**Comment by saz33m1** [comment]
@mikesealey @mike12345567  @joebudi Can someone from the team please answer this question? I can see my public form screen, but can't submit to a private table.  This is basic functionality available with all your competitors I have tested so far.
<img width="922" alt="image" src="https://github.com/user-attachments/assets/cf388f85-70f3-4983-81aa-1621acf35f41">

**Reply by mike12345567** [reply]
Hi @saz33m1 - this is something we're currently discussing, however one thing we recommend is using automations to create/update rows from a public screen. Creating public write tables can be quite dangerous, but using an automation offers an awful lot more control. You can use an automation with an app action trigger, which is set to allow public execution - this can then be triggered from a button action. If you need information from the row creation you can use sync automations to return information from the row create/update call.

**Reply by saz33m1** [reply]
@mike12345567 I tried it, but having some trouble. The issue is that my button is not even showing when I try to load the form. Can you record a short tutorial please? total n00b here.
